display/app/controllers/application_controller.rb

```diff
@@ -1265,9 +1266,13 @@ def ci_image_url(ci)
   end
 
   def ci_class_image_url(ci_class_name)
-    split = ci_class_name.split('.')
-    split = split[1..-1] if split.first == 'mgmt'
-    "#{asset_url_prefix}#{split[1..-1].join('.')}/#{split.last}.png"
+    if Cms::CiMd.look_up!(ci_class_name)
```

Comment:
@lkhusid Isn't this a perf concern to call cms for each image URL construction? Earlier, it was just a string interpolation, now a network i/o involved here.